natsstreaming/nats_streaming.go

```diff
@@ -25,24 +25,40 @@ const (
 	OffsetNewest int64 = -1
 )
 
+// ConnectionTimeOutConfig encapsulates interval and timeout seconds fields
+// used for configuring nats streaming connection loss detection
+type ConnectionTimeOutConfig struct {
+	// Number of seconds to wait for a response from nats streaming
+	Seconds int
+	// Number of attempts to establish a connection
+	Tries int
+}
+
 // AsyncMessageSinkConfig is the configuration parameters for an
 // AsyncMessageSink.
 type AsyncMessageSinkConfig struct {
 	URL       string
 	ClusterID string
 	ClientID  string
 	Subject   string
+	TimeOut   ConnectionTimeOutConfig
 }
 
 func NewAsyncMessageSink(config AsyncMessageSinkConfig) (substrate.AsyncMessageSink, error) {
-	sink := asyncMessageSink{subject: config.Subject}
+	sink := asyncMessageSink{subject: config.Subject, connectionLost: make(chan error, 1)}
 
 	clientID := config.ClientID
 	if clientID == "" {
 		clientID = generateID()
 	}
 
-	sc, err := stan.Connect(config.ClusterID, clientID, stan.NatsURL(config.URL))
+	sc, err := stan.Connect(
+		config.ClusterID, clientID, stan.NatsURL(config.URL),
+		stan.Pings(config.TimeOut.Seconds, config.TimeOut.Tries),
```

Comment:
Should `config.TimeOut.Seconds` and `config.TimeOut.Tries` have some default values is it ok for them to be 0?

Reply:
they should have defaults, but instantiation will fail if incorrectly configured. happy to add for convenience.